src/tests/App.test.tsx

```diff
@@ -0,0 +1,94 @@
+import { screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { vi } from 'vitest';
+import { renderWithRedux } from './helpers/renderWith';
+import * as APIModule from '../utils/fetchApi';
+import { News } from '../utils/types';
+import getImageURL from '../utils/getImageURL';
+import { mockNews, mockBreakingNews } from './helpers/mockData';
+import { BASE_IMAGE_URL } from '../components/NewHero/NewHero';
+import App from '../App';
+
+const MOCK_INITIAL_STATE = {
+  news: {
+    status: 'Requisição concluída',
+    news: mockNews,
+    breakingNews: mockBreakingNews,
+    filter: '',
+    quantity: 12,
+    toggleOrientationType: true,
+    activeButton: 'maisRecentes',
+    content: true,
+  },
+};
+
+beforeEach(() => {
+  vi.spyOn(APIModule, 'fetchApi').mockResolvedValue(mockNews as News);
+});
+
+afterEach(() => {
+  vi.clearAllMocks();
+});
+
+const user = userEvent.setup();
+
+describe('Testes da aplicação', () => {
+  test('Verificar o Header', async () => {
+    renderWithRedux(<App />);
+    const header = screen.getByTestId('header');
```

Comment:
[A documentação do RTL](https://testing-library.com/docs/queries/about#priority) recomenda que sejam utilizados os `data-testid` apenas se nenhuma das outras opções for possível - o `getByRole` sendo a melhor opção possível.

Como está tentando buscar um `header` aqui, [pode buscá-lo pela sua `role`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/banner_role) e eliminar o `data-testid`

```suggestion
    const header = screen.getByRole('banner');
```